packages/core/src/config.ts

```diff
@@ -259,6 +259,7 @@ const composeAutoExtensionConfig = (
       output: {
         filename: {
           js: `[name]${jsExtension}`,
+          ...config.output?.filename,
```

Comment:
What about early return if `autoExtension` is disabled, and the `config.output?.filename` won't be overridden.

Reply:
Maybe the commit message is not accurate, users can modify `config.output?.filename` no matter `autoExtension` enabled or not. They can customize both filename and extname.

Reply:
I will add more tests to cover it.